### PR TITLE
add unleash package

### PR DESCRIPTION
**Pull request by ANKDGIRI777**
Added function to pass ENV variables

### CODE REVIEW

File: .yarnrc [hunk 1]
```diff
@@ -0,0 +1,2 @@
+// .yarnrc
+registry "http://192.168.0.142:8473"
```
Comment (prathap-safe) [inline .yarnrc:2]: Remove once tested.

File: package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "unleash-safe-security",
-  "version": "1.0.0",
+  "version": "1.0.12",
```
Comment (prathap-safe) [inline package.json:3]: Should be revisited before merge to reflect the correct version.

File: src/index.ts [hunk 3]
```diff
@@ -1,42 +1,40 @@
...
-import { initialize, Strategy } from "unleash-client";
-
-/**
- * The unleash library only supports a class
- * as a strategy, there is no workaround here,
- * have to disable this for this particular scenario
- */
-// eslint-disable-next-line functional/no-class
+import { Strategy, initialize, isEnabled, Unleash } from 'unleash-client';
+const logger=console;
```
Comment (prathap-safe) [inline src/index.ts:4]: Remove this & its usage before merge.

File: src/index.ts [hunk 4]
```diff
@@ -1,42 +1,40 @@
...
     }
 
     isEnabled(parameters: { baseUrl: string }): boolean {
         const allowedList = new Set(
             parameters.baseUrl.split(",").map(url => url.trim().toLowerCase())
         );
-
-        const baseUrl = (process.env.BASE_URL as string).toLowerCase();
+        logger.log("In isEnabled function");
+        const baseUrl = (this.baseUrl as string).toLowerCase();
```
Comment (prathap-safe) [inline src/index.ts:19]: It is not advisable to use the same variable name in different scopes.

File: src/index.ts [hunk 5]
```diff
@@ -1,42 +1,40 @@
...
- * as a strategy, there is no workaround here,
- * have to disable this for this particular scenario
- */
-// eslint-disable-next-line functional/no-class
+import { Strategy, initialize, isEnabled, Unleash } from 'unleash-client';
+const logger=console;
 class BaseUrlStrategy extends Strategy {
-    constructor() {
-        // eslint-disable-next-line @typescript-eslint/no-unsafe-call
+    baseUrl: string ;
```
Comment (prathap-safe) [inline src/index.ts:6]: Rename to `url`

File: src/index.ts [hunk 6]
```diff
@@ -1,42 +1,40 @@
+type Arg=string | undefined;
```
Comment (prathap-safe) [inline src/index.ts:1]: Not needed; you can use `?` (optional) instead in the function.

**Review by prathap-safe — CHANGES_REQUESTED** [review]

File: src/index.ts (hunk 3 above)
Comment (ANKDGIRI777) [reply src/index.ts:4]: Yes, will be kept till branch is reviewed completely

File: src/index.ts (hunk 4 above)
Comment (ANKDGIRI777) [reply src/index.ts:19]: Noted, and changed

File: src/index.ts (hunk 5 above)
Comment (ANKDGIRI777) [reply src/index.ts:6]: Renamed

**Pull request closed without merging**